src/tokenizer.rs

```diff
@@ -96,13 +114,20 @@ pub fn tokenize<'a>(source: &'a str) -> Result<Vec<Token<'a>>, TokenizeError<'a>
             .or_else(|| try_advance(current, &PATTERN_CLOSE_PAREN, |_| TokenKind::CloseParen));
 
         match result {
-            Some((next_current, token_kind)) => {
+            Some((next_current, eaten_str, token_kind)) => {
                 current = next_current;
 
                 tokens.push(Token {
                     whitespace,
                     kind: token_kind,
+                    line: current_line,
+                    column: current_column,
                 });
+
+                // This will cause problems if tokens match more than one line,
+                // but that seems really really unlikely - that would mean
+                // allowing whitespace in the tokens!
+                current_column += eaten_str.len();
```

Comment:
We could add a parameter to Token that indicates the number of lines the token spanned. I think that multi-line strings are important enough to get right for preservation!